dali/pipeline/operator/op_schema.h

```diff
@@ -572,29 +626,55 @@ class DLL_PUBLIC OpSchema {
     return result.str();
   }
 
-  DLL_PUBLIC std::string GetInputName(int input_idx) {
+  struct InOutDoc {
+    std::string name = {};
+    std::string type_doc = {};
+    std::string doc = {};
+  };
+
+  DLL_PUBLIC InOutDoc GetPerInputDoc(int input_idx) {
     CheckInputIndex(input_idx);
-    DALI_ENFORCE(HasInputDox(),
+    DALI_ENFORCE(HasPerInputDoc(),
                  "Input documentation was not specified for this operator.");
     DALI_ENFORCE(!input_dox_[input_idx].name.empty(),
                  make_string("Docstring for input ", input_idx,
                              "was not set. All inputs should be documented."));
-    return input_dox_[input_idx].name;
-  }
-
-  DLL_PUBLIC std::string GetInputType(int input_idx) {
-    CheckInputIndex(input_idx);
-    DALI_ENFORCE(HasInputDox(),
-                 "Input documentation was not specified for this operator.");
-    return input_dox_[input_idx].type_doc;
-  }
-
-  DLL_PUBLIC std::string GetInputDox(int input_idx) {
-    CheckInputIndex(input_idx);
-    DALI_ENFORCE(HasInputDox(),
-                 "Input documentation was not specified for this operator.");
-    return input_dox_[input_idx].doc;
-  }
+    return input_dox_[input_idx];
+  }
+
+  DLL_PUBLIC InOutDoc GetPerOutputDoc(int output_idx) {
+    CheckOutputIndex(output_idx);
+    DALI_ENFORCE(HasPerOutputDoc(),
+                 "Output documentation was not specified for this operator.");
+    DALI_ENFORCE(!output_dox_[output_idx].name.empty(),
+                 make_string("Docstring for output ", output_idx,
+                             "was not set. All outputs should be documented."));
+    return output_dox_[output_idx];
+  }
+
+  // DLL_PUBLIC std::string GetPerInputName(int input_idx) {
+  //   CheckInputIndex(input_idx);
+  //   DALI_ENFORCE(HasPerInputDoc(),
+  //                "Input documentation was not specified for this operator.");
+  //   DALI_ENFORCE(!input_dox_[input_idx].name.empty(),
+  //                make_string("Docstring for input ", input_idx,
+  //                            "was not set. All inputs should be documented."));
+  //   return input_dox_[input_idx].name;
+  // }
+
+  // DLL_PUBLIC std::string GetPerInputType(int input_idx) {
+  //   CheckInputIndex(input_idx);
+  //   DALI_ENFORCE(HasPerInputDoc(),
+  //                "Input documentation was not specified for this operator.");
+  //   return input_dox_[input_idx].type_doc;
+  // }
+
+  // DLL_PUBLIC std::string GetPerInputDoc(int input_idx) {
+  //   CheckInputIndex(input_idx);
+  //   DALI_ENFORCE(HasPerInputDoc(),
+  //                "Input documentation was not specified for this operator.");
+  //   return input_dox_[input_idx].doc;
+  // }
```

Comment:
?

Reply:
Yeah, I will remove it, I'm still working on some parts and wanted to render the docs for everyone to see.